libp2p-comit/src/handler.rs

```diff
@@ -90,6 +90,12 @@ pub enum ProtocolInEvent {
     Message(OutboundMessage),
 }
 
+impl Clone for ProtocolInEvent {
+    fn clone(&self) -> Self {
+        unimplemented!("we cannot implement clone because we have channels in the messages")
```

Comment:
Yes, this is concerning. However, our e2e tests pass with this so I can only assume that libp2p actually never calls `.clone()` on our events.

I would argue that this is therefore safe to use, esp. because it is contained in our "legacy" networking code and will be removed soonish!

That being said, if people feel strongly about this, we can:

a) refactor the old code to not use channels (something I thought of doing for a while but never seemed worth it as we move away from it anyway)
b) wait with this dependency update

Opinions please :)

cc @tcharding @bonomat @D4nte

Reply:
Should it be `unreachable!` instead?

Reply:
I don't think it is unreachable :sweat_smile:

Reply:
I just don't know under which conditions `libp2p` tries to clone the events :o

Reply:
No strong opinion on this. I guess if it ever fails we can fix it then, or revert these changes. It's an interesting situation though :stuck_out_tongue: Thanks for asking!

Reply:
You probably have a better insight into this than be. I'm fine with having this `unimplemented()` in here and wait until this becomes a problem. If not, then it won't hurt neither :)

Reply:
Yes I thought the same way :)

Just wanted to make everyone aware not sneak something like this in :D